### ISSUE TITLE
Support for Amazon SQS Extended Client

### ISSUE THREAD

**Issue by Rajneesh-Testsigma** [JaidenAshmore/java-dynamic-sqs-listener#382]
Can this library support extended client . Since with SQSAsycClient we can use only 256 KB of message ..... I have requirement to process the message of size more that 256 KB which  Amazon SQS Extended Client do that can it be supported there ?

**Comment by JaidenAshmore** [comment]
Hey, looking at this [repo](https://github.com/awslabs/amazon-sqs-java-extended-client-lib/) it looks like the implementation of this only uses the synchronous version of the SqsClient whereas this library is very heavily dependent on the asynchronous SqsAsyncClient interface. So unfortunately we can't just easily dump it in and have it work.

Doing some research, e.g. [this issue](https://github.com/awslabs/amazon-sqs-java-extended-client-lib/issues/5) started my journey, it appears that someone has implemented an SqsAsyncClient for this extended library: https://github.com/Harsha-changejar/amazon-sqs-java-extended-client-lib. It looks to only be a fork and doesn't have a published version in the repository though so I dunno if you would be able to use that.

Ultimately, from this library's perspective, I wouldn't try and support both the async and the non-async version and so the next best step for you would be to try to find (or implement) a `SqsAsyncClient` implementation that does this extended functionality.

Let me know if that doesn't sound right!

**Comment by Rajneesh-Testsigma** [comment]
Thank for response... I had written something sort of like this but still it was helpful.

**Comment by JaidenAshmore** [comment]
Yeah so if you can get that Extended Async client implementation running with your code it shoouuuuld work. If it doesn't let me know and I can dig into it further.
